engine/src/lib.rs

```diff
@@ -1,143 +1,49 @@
-use eyre::Result;
-use handler::SlotHandler;
-use info::L2Info;
-use ledger::Ledger;
-use lumio_types::{
-    p2p::{SlotArtifact, SlotAttribute},
-    Slot,
-};
-use std::sync::{
-    atomic::{AtomicBool, Ordering},
-    Arc,
-};
-use tokio::sync::mpsc::{Receiver, Sender};
-
 pub mod handler;
 pub mod info;
 pub mod ledger;
 pub mod sub;
 
-#[derive(Clone)]
-pub struct EngineService<L> {
-    ledger: Arc<L>,
-    sync_task_token: TaskToken,
-}
-
-impl<L: Ledger + Send + Sync + 'static> EngineService<L> {
-    pub fn new(ledger: L) -> Self {
-        let ledger = Arc::new(ledger);
-        Self {
-            ledger,
-            sync_task_token: TaskToken::default(),
+use crate::{handler::SlotHandler, ledger::Ledger};
+use eyre::Result;
+use lumio_p2p::{Config, Node};
+use std::sync::Arc;
+
+pub async fn spawn_engine_service<L: Ledger + Send + Sync + 'static>(
+    ledger: L,
+    p2p_cfg: Config,
+    engine_type: EngineType,
+) -> Result<()> {
+    let ledger = Arc::new(ledger);
+    let (mut node, runner) = Node::new(p2p_cfg)?;
+    tokio::spawn(runner.run());
```

Comment:
Lets move this part into the op-sol or op-move itself

Reply:
make sense